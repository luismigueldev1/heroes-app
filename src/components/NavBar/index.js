import React, { useContext } from "react";
import { Link, NavLink, useHistory } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

function NavBar() {
  const history = useHistory();
  const { user, dispatch } = useContext(AuthContext);

  const handleLogout = () => {
    dispatch({
      type: types.logout,
    });
    history.replace("/login");
  };
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <Link to="/" className="navbar-brand">
          HeroesApp
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNav"
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav me-auto">
            <li className="nav-item ">
              <NavLink
                exact
                to="/dc"
                activeClassName="active"
                className="nav-link"
              >
                DC
              </NavLink>
            </li>
            <li className="nav-item ">
              <NavLink
                exact
                to="/marvel"
                activeClassName="active"
                className="nav-link"
              >
                Marvel
              </NavLink>
            </li>
            <li className="nav-item ">
              <NavLink
                exact
                to="/search"
                activeClassName="active"
                className="nav-link"
              >
                Search
              </NavLink>
            </li>
          </ul>
          <ul className="navbar-nav">
            <li className="nav-item nav-link">
              <span className="text-info">{user.name}</span>
            </li>

            <li className="nav-item">
              <button className="btn btn-secondary" onClick={handleLogout}>
                Logout
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}

export default NavBar;
