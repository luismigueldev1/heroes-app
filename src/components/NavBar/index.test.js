import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router-dom";
import NavBar from ".";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

describe("Pruebas en <Navbar/>", () => {
  const historyMock = {
    push: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
    replace: jest.fn(),
  };
  const contextValues = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Luis Miguel",
    },
  };
  const wrapper = mount(
    <AuthContext.Provider value={contextValues}>
      <MemoryRouter>
        <Router history={historyMock}>
          <NavBar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("debe de mostrarse correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe(
      contextValues.user.name
    );
  });

  test("debe de llamar el logout y usar history", () => {
    wrapper.find(".btn-secondary").simulate("click");

    expect(contextValues.dispatch).toHaveBeenCalledWith({ type: types.logout });
    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });
});
