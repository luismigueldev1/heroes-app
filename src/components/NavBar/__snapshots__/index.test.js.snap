// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar/> debe de mostrarse correctamente  1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      HeroesApp
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav me-auto"
      >
        <li
          className="nav-item "
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            DC
          </a>
        </li>
        <li
          className="nav-item "
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Marvel
          </a>
        </li>
        <li
          className="nav-item "
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item nav-link"
        >
          <span
            className="text-info"
          >
            Luis Miguel
          </span>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
